src/java.base/share/classes/java/lang/foreign/MemorySession.java

```diff
@@ -127,157 +78,44 @@
  * @since 19
  */
 @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
-public sealed interface MemorySession extends AutoCloseable, SegmentAllocator permits MemorySessionImpl, MemorySessionImpl.NonCloseableView {
+public sealed interface MemorySession permits MemorySessionImpl {
 
     /**
      * {@return {@code true}, if this memory session is alive}
      */
     boolean isAlive();
 
     /**
-     * {@return {@code true}, if this session is a closeable memory session}.
+     * {@return {@code true} if the provided thread can access and/or obtain segments associated with this memory session}
```

Comment:
Is the following accurate and more concise?
```suggestion
     * {@return {@code true} if the provided thread can access and/or associate segments with this memory session}
```